### PR TITLE
Creating Simpler Nested Proto Parse Templates

### PR DESCRIPTION
**Pull request by AsherGlick**
The previous logic had duplicated code chunks. The new logic is much more flexible and unifies the template branches to remove duplicate chunks.

### CODE REVIEW

File: xml_converter/generators/cpp_templates/class_template.cpp [hunk 1]
```diff
@@ -149,55 +149,28 @@ waypoint::{{cpp_class}} {{cpp_class}}::as_protobuf() const {
...
-                        this->{{attribute_variable.attribute_name}} = proto_{{cpp_class_header}}.{{attribute_variable.protobuf_field}}();
-                        this->{{attribute_variable.attribute_flag_name}} = true;
-                    }
-                {% endif %}
+            {% if (attribute_variable.attribute_type in ["MultiflagValue", "CompoundValue", "Custom", "CompoundCustomClass"]) %}
+                if (proto_{{cpp_class_header}}{{attribute_variable.proto_drilldown_calls}}.has_{{attribute_variable.protobuf_field}}()) {
+                    this->{{attribute_variable.attribute_name}} = from_proto_{{attribute_variable.class_name}}(proto_{{cpp_class_header}}{{attribute_variable.proto_drilldown_calls}}.{{attribute_variable.protobuf_field}}());
+                    this->{{attribute_variable.attribute_flag_name}} = true;
+                }
+            {% elif (attribute_variable.class_name == "string") %}{# TODO: why is this .class_name when the others are .attribute_name #}
```
Comment (klingbolt) [inline xml_converter/generators/cpp_templates/class_template.cpp:159]: I think this TODO is meant to say `attribute_type` instead of name. I think it can be changed to `attribute_type` without issue since `class_name` is directly assigned by `attribute_type` in the generator for String types. Can be a separate PR

**Review by klingbolt — APPROVED** [review]

File: xml_converter/generators/cpp_templates/class_template.cpp (hunk 1 above)
Comment (AsherGlick) [reply xml_converter/generators/cpp_templates/class_template.cpp:159]: Yes it is supposed to be `attribute_type`. It will changed in a later PR.
